www/caddy/src/opnsense/mvc/app/models/OPNsense/Caddy/Caddy.php

```diff
@@ -107,6 +108,52 @@ private function checkSubdomainsAgainstDomains($subdomains, $domains, $messages)
         }
     }
 
+    // Get the OPNsense WebGUI configuration and store it
+    private $webgui;
+
+    private function getWebGui() {
+        if (!$this->webgui) {
+            $this->webgui = Config::getInstance()->object()->system->webgui ?? null;
+        }
+        return $this->webgui;
+    }
+
+    // Get the default OPNsense WebGUI ports
+    private function getWebGuiPorts() {
+        $webgui = $this->getWebGui();
```

Comment:
Best move the config access inside the function ( `getWebGui()` isn't used anywhere else)
```
$webgui = Config::getInstance()->object()->system->webgui ?? null;
```